server/built_in_tools.go

```diff
@@ -38,6 +38,13 @@ func (p *Plugin) toolResolveLookupMattermostUser(context ai.ConversationContext,
 		return "", errors.Wrap(err, "failed to lookup user")
 	}
 
+	// If this isn't a DM fail for users who are not SSO users.
+	if !context.IsDMWithBot() {
+		if user.AuthService != model.UserAuthServiceSaml && user.AuthService != model.UserAuthServiceLdap {
```

Comment:
At least with LDAP it looks like switching from email auth does not immediately update the profile, meaning you could have `user.AuthService == model.UserAuthServiceLdap` and user attributes that were manually set while on email auth. I *think* that's the same for SAML too, but I need to confirm.

Anyway, this check isn't good enough.

Reply:
Don't see a way to check for this case. Removing the exception for now.